src/storage/mvcc/txn.rs

```diff
@@ -301,7 +301,7 @@ impl<S: Snapshot> MvccTxn<S> {
                 let max_read_ts = self.concurrency_manager.max_read_ts();
                 let min_commit_ts =
                     cmp::max(cmp::max(max_read_ts, self.start_ts), for_update_ts).next();
-                lock.min_commit_ts = min_commit_ts;
+                lock.min_commit_ts = cmp::max(lock.min_commit_ts, min_commit_ts);
```

Comment:
The `min_commit_ts` must be greater than or equal to `lock.min_commit_ts`. But I'm fine with it.

Reply:
I'm actually not very sure. The problem is whether the `max_read_ts` need to be pushed with the `min_commit_ts` from the request. If so, the final calculated commit_ts may be `min_commit_ts + 1`.

Reply:
And if the min_commit_ts from request is a calculated ts, then the final commit_ts may violate this https://github.com/tikv/sig-transaction/issues/21

Reply:
@sticnarf PTAL

Reply:
Now only pessimistic async-commit transaction's `min_commit_ts` is not zero before prewriting and it equals `lock.for_update_ts + 1`. It's never pushed because only when the transaction has prewritten some locks, it's possible the transaction blocks read. But in design, these locks' `min_commit_ts` are zero, so the client can't push it.

Reply:
In this PR https://github.com/tikv/tikv/pull/8395, even the min_commit_ts of the pessimistic lock of a pessimistic transaction cannot be pushed forward. I think `lock.min_commit_ts` must be 0 for an async commit txn?

Reply:
> And if the min_commit_ts from request is a calculated ts, then the final commit_ts may violate this tikv/sig-transaction#21
Yes, it's an unresolved problem stopping us from using a calculated TS to read. But I think it's unrelated to this PR.